File: src/main/java/kingdoms/server/handlers/packets/server/SPacketBank.java
package kingdoms.server.handlers.packets.server;

import kingdoms.api.network.AbstractPacket;
import kingdoms.server.PlayerProvider;
import net.minecraft.entity.player.EntityPlayerMP;

public final class SPacketBank extends AbstractPacket
{
    public SPacketBank(){}
    public SPacketBank(short count, boolean deposit)
    {
        buf().writeShort(count);
        buf().writeBoolean(deposit);
    }

    @Override
    public void server(EntityPlayerMP player)
    {
        PlayerProvider provider = PlayerProvider.get(player);

        int count = buf().readShort();
        boolean deposit = buf().readBoolean();

        if (deposit)
        {
            if (provider.getGoldTotal() >= count)
            {
                provider.addBankGold(count);
                provider.decreaseGold(count);
            }
        }
        else {
            if (provider.getBankGold() >= count)
            {
                provider.addGold(count);
                provider.decreaseBankGold(count, player);
            }
        }
    }
}

File: src/main/java/kingdoms/server/handlers/packets/server/SPacketBank.kt
package kingdoms.server.handlers.packets.server

import kingdoms.api.network.SimplePacket
import kingdoms.server.PlayerProvider
import net.minecraft.entity.player.EntityPlayerMP

class SPacketBank(): SimplePacket()
{
    constructor(count: Short, deposit: Boolean): this()
    {
        buf().writeShort(count.toInt())
        buf().writeBoolean(deposit)
    }

    override fun server(player: EntityPlayerMP)
    {
        val provider = PlayerProvider.get(player)

        val count = buf().readShort().toInt()
        val deposit = buf().readBoolean()

        if (deposit)
        {
            if (provider.goldTotal >= count)
            {
                provider.addBankGold(count)
                provider.decreaseGold(count)
            }
        }
        else
        {
            if (provider.bankGold >= count)
            {
                provider.addGold(count)
                provider.decreaseBankGold(count)
            }
        }
    }
}